tutorials/llm/nemo2-peft.ipynb

```diff
@@ -0,0 +1,2052 @@
+    "# Learning Goals\n",
+    "\n",
+    "## Parameter-Efficient Fine-Tuning (PEFT)\n",
+    "\n",
+    "This notebook aims to demonstrate how to adapt or customize foundation models to improve performance on specific tasks using NeMo 2.0.\n",
+    "\n",
+    "This optimization process is known as fine-tuning, which involves adjusting the weights of a pre-trained foundation model with custom data.\n",
+    "\n",
+    "Considering that foundation models can be significantly large, a variant of fine-tuning has gained traction recently known as PEFT. PEFT encompasses several methods, including P-Tuning, LoRA, Adapters, IA3, etc. NeMo 2.0 currently supports Low-Rank Adaptation (LoRA) method.\n",
+    "\n",
+    "This playbook involves applying LoRA to Llama3 using NeMo 2.0. \n",
+    "\n",
+    "## NeMo 2.0\n",
+    "\n",
+    "In NeMo 1.0, the main interface for configuring experiments is through YAML files. This approach allows for a declarative way to set up experiments, but it has limitations in terms of flexibility and programmatic control. NeMo 2.0 is an update on the NeMo Framework which introduces several significant improvements over its predecessor, NeMo 1.0, enhancing flexibility, performance, and scalability.\n",
+    "\n",
+    "- Python-Based Configuration - NeMo 2.0 transitions from YAML files to a Python-based configuration, providing more flexibility and control. This shift makes it easier to extend and customize configurations programmatically.\n",
+    "\n",
+    "- Modular Abstractions - By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 simplifies adaptation and experimentation. This modular approach allows developers to more easily modify and experiment with different components of their models.\n",
+    "\n",
+    "- Scalability - NeMo 2.0 seamlessly scales large-scale experiments across thousands of GPUs using NeMo-Run, a powerful tool designed to streamline the configuration, execution, and management of machine learning experiments across computing environments.\n",
+    "\n",
+    "By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 makes it easy for users to adapt the framework to their specific use cases and experiment with various configurations. This section offers an overview of the new features in NeMo 2.0 and includes a migration guide with step-by-step instructions for transitioning your models from NeMo 1.0 to NeMo 2.0.\n",
+    "\n",
+    "## NeMo-Run\n",
+    "\n",
+    "NeMo-Run is a powerful tool designed to streamline the configuration, execution, and management of machine learning experiments across various computing environments. To run your experiments with Nemo-Run, you need to follow the three steps:\n",
+    "1. Configure your function\n",
+    "\n",
+    "2. Define your Executor\n",
+    "\n",
+    "3. Run your experiment\n",
+    "\n",
+    "We will be using NeMo-Run to run our experiments in this notebook.\n",
+    "\n",
+    "\n",
+    "# NeMo Tools and Resources\n",
```

Comment:
Please move NeMo Tool and Resources, and Educational Resources at the bottom

Reply:
Done, thanks!